realm/realm-library/src/main/java/io/realm/RealmCache.java

```diff
@@ -504,4 +514,14 @@ private int getTotalGlobalRefCount() {
 
         return totalRefCount;
     }
+
+    /**
+     * If a Realm instance is GCed but `Realm.close()` is not called before, we still want to track the cache for
+     * debugging. Adding them to the list to keep the strong ref of the cache to prevent the cache gets GCed.
+     */
+    void leak() {
+        if (!isLeaked.getAndSet(true)) {
+            leakedCaches.add(this);
```

Comment:
The following is my understanding. Is that correct?

If a user writes code below, `leakedCaches` keeps a strong reference to the `RealmCache` instance and it never release it. Of course the referenced `RealmCache` instance never be GCed, but it works as expected (== Realm instance in it is GCed) even if this kind of leak happens.

```
Realm.getDefaultInstance();
System.gc(); // leak() is called.
Realm r = Realm.getDefaultInstance();
r.close();
r.close();
```